### ISSUE TITLE
not all subtemplates/inherited templates are extended (#123 re-open)

### ISSUE THREAD

**Issue by chilek** [smarty-php/smarty#183]
This is #123 issue continuation ;-)
Demo presenting the problem https://www.chilan.com/smarty-test2/
Source code: https://www.chilan.com/smarty-3.1.29-extendsall-test.tar.xz

**Comment by uwetews** [comment]
You have a very special use case.
The problem are the {include} tags within {block}.
{include} tags which extend inheritance templates will in the normal case start a new inheritance root.
This is a feature with allows to use different template inheritance chains. Otherwise it could also break
the inheritance structure.
This seems to be not what you did intend by setting the default resource type to extendsall.

Could you for a test insert the following lines in resource.extendsall.php at the end of method populate() and give a feedback if that resolves your problem.

```php
        if (isset($_template->ext->_inheritance) && $_template->ext->_inheritance->state == 3) {
            $_template->ext->_inheritance->state = 2;
        }
```

**Comment by uwetews** [comment]
The fix is now in the master branch and will later be included in 3.1.30

I hope it's now finally done ...

**Comment by chilek** [comment]
Thanks a lot, Uwe! It looks like it has resolved my test case problem.
